lib/philomena/images.ex

```diff
@@ -427,6 +431,44 @@ defmodule Philomena.Images do
         {:ok, count}
     end)
     |> Repo.transaction()
+    |> case do
+      {:ok, %{image: {image, _added, _removed}}} = res ->
+        update_tag_change_limits_after_commit(image, attribution)
+
+        res
+
+      err ->
+        err
+    end
+  end
+
+  defp check_tag_change_limits_before_commit(image, attribution) do
+    user = attribution[:user]
+    ip = attribution[:ip]
+
+    cond do
+      Limits.is_limited_for_tag_count?(user, ip) ->
+        {:error, "Limit exceeded"}
+
+      Limits.is_limited_for_rating_count?(user, ip) ->
+        {:error, "Limit exceeded"}
+
+      length(image.added_tags) + length(image.removed_tags) > 1000 ->
```

Comment:
1000 seems a bit too generous, maybe 100?

Reply:
I think 100 might be a bit too aggressive since this blocks all tag changes with batch sizes of greater than 100. (Also it even applies for mods, which I should fix...)